spec/services/order_invoice_comparator_spec.rb

```diff
@@ -2,387 +2,399 @@
+  before do
+    Spree::Order.where(id: order.id).update_all(payment_total: order.payment_total + 10)
+  end
 
-    context "changes on the order object" do
-      describe "detecting relevant attribute changes" do
-        it "returns true if a relevant attribute changes" do
-          Spree::Order.where(id: order.id).update_all(total: order.total + 10)
-          order.reload
-          expect(subject).to be true
-        end
+  it "returns #{boolean} if a #{type} attribute changes" do
+    order.reload
+    expect(subject).to be boolean
+  end
+end
 
-        context "additional tax total changes" do
-          let(:order) do
-            create(:order_with_taxes, product_price: 110, tax_rate_amount: 0.1,
-                                      included_in_price: false)
-              .tap do |order|
-              order.create_tax_charge!
-              order.update_shipping_fees!
-            end
-          end
-
-          it "returns true" do
-            Spree::TaxRate.first.update!(amount: 0.15)
-            order.create_tax_charge!
-            order.reload
-            expect(subject).to be true
-          end
-        end
+shared_examples "attribute changes - order total" do |boolean, type|
+  before do
+    Spree::Order.where(id: order.id).update_all(total: order.total + 10)
+  end
 
-        context "included tax total changes" do
-          let(:order) do
-            create(:order_with_taxes, product_price: 110, tax_rate_amount: 0.1,
-                                      included_in_price: true)
-              .tap do |order|
-              order.create_tax_charge!
-              order.update_shipping_fees!
-            end
-          end
-
-          it "returns true" do
-            Spree::TaxRate.first.update!(amount: 0.15)
-            order.create_tax_charge!
-            order.reload
-            expect(subject).to be true
-          end
-        end
+  it "returns #{boolean} if a #{type} attribute changes" do
+    order.reload
+    expect(subject).to be boolean
+  end
+end
 
-        context "shipping method changes" do
-          let(:shipping_method) { create(:shipping_method) }
-          it "returns true" do
-            Spree::ShippingRate.first.update(shipping_method_id: shipping_method.id)
-            expect(subject).to be true
-          end
-        end
+shared_examples "attribute changes - order state: cancelled" do |boolean, type|
+  before do
+    order.cancel!
+  end
+
+  it "returns #{boolean} if a #{type} attribute changes" do
+    order.reload
+    expect(subject).to be boolean
+  end
+end
+
+shared_examples "attribute changes - tax total changes" do |boolean, type, included_boolean|
+  let(:order) do
+    create(:order_with_taxes, product_price: 110, tax_rate_amount: 0.1,
+                              included_in_price: included_boolean)
+      .tap do |order|
+      order.create_tax_charge!
+      order.update_shipping_fees!
+    end
+  end
+
+  context "if included is #{included_boolean}" do
```

Comment:
It might be a bit clearer if the full attribute name is included here: 

```suggestion
  context "if included_in_price is #{included_boolean}" do
```

Reply:
Renamed the description :ok: